sql/src/main/java/io/crate/operation/scalar/string/HashFunctions.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar.string;
+
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.operation.scalar.ScalarFunctionModule;
+import io.crate.operation.scalar.UnaryScalar;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.apache.lucene.util.BytesRef;
+
+import java.security.DigestException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+public final class HashFunctions {
+
+    private static final List<DataType> SUPPORTED_INPUT_TYPE = Collections.singletonList((DataTypes.STRING));
+
+    private enum HashMethod {
+        MD5("MD5"),
+        SHA1("SHA1"),
+        SHA256("SHA-256"),
+        SHA512("SHA-512");
+
+        private final String method;
+
+        HashMethod(String method) {
+            this.method = method;
+        }
+
+        public BytesRef digest(BytesRef input) {
+            final MessageDigest messageDigest;
+            try {
+                messageDigest = MessageDigest.getInstance(method);
```

Comment:
`MessageDigests` doesn't have a function for SHA512. I've removed SHA512. We could add the function to `MessageDigests` in an upstream patch.